### PR TITLE
Assignment8 Submission - Raymond Shiner

### PR DESCRIPTION
**Pull request by raymondshiner**

### CODE REVIEW

**Comment by raymondshiner** [comment]
I just Realized that on line 92 of Mainwindow.Xaml.cs, i assign StartTime, but it never gets used. This is because i forgot to add to the string on lines 98 and 99 of the same file(when the task completes) this code 

97 - DateTime now = TimeManager.Now();
98 - DateTime interval = now - StartTime;
99 (Append to End of String) - " - Task was completed in interval seconds".

Basically i meant to in that method utilize startTime to also list the interval. Could've done it in seconds just forgot cuz i was frantically trying to get my TimeManager class to work. Don't expect a grade change or anything just trying to explain my intention for what ended up being useless code.

File: Assignment8/Assignment8/TimeManager.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        public Action StartStopButtonClick { get; set; }
+        public Action DeleteButtonClick { get; set; }
+
+        public void RaiseEventStartStop(object sender, RoutedEventArgs routedEventArgs)
+        {
+            StartStopButtonClick?.Invoke();
+        }
+
+        public void RaiseEventDelete(object sender, RoutedEventArgs e)
```
Comment (Keboo) [inline Assignment8/Assignment8/TimeManager.cs]: The delete functionality didn't need to go through the TimeManager and would have been simpler to do in the MainWindow directly.

File: Assignment8/Assignment8/TimeManager.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using System.Windows;
+
+namespace Assignment8
+{
+    public class TimeManager : IDateTime
+    {
+        public Action StartStopButtonClick { get; set; }
+        public Action DeleteButtonClick { get; set; }
+
+        public void RaiseEventStartStop(object sender, RoutedEventArgs routedEventArgs)
```
Comment (Keboo) [inline Assignment8/Assignment8/TimeManager.cs]: I would have expected this method to do things like keep track of the timer state (basically what your `ThisClickIsStart` boolean in MainWindow is doing).

File: Assignment8/Assignment8/MainWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            _timer.Start();
+            TaskRunningMessage.Content = "Press Start to Start a Task";
+            ThisClickIsStart = false;
+
+            var manager = new TimeManager();
+            manager.StartStopButtonClick += StartStop_OnClick_SetCurrentClick;
+            manager.StartStopButtonClick += Stop_OnClick_DisplayTaskFinished;
+            manager.StartStopButtonClick += Stop_OnClick_EndTask;
+            manager.StartStopButtonClick += Start_OnClick_DisplayTaskRunning;
+            manager.StartStopButtonClick += Start_OnClick_StartTask;
```
Comment (Keboo) [inline Assignment8/Assignment8/MainWindow.xaml.cs]: Though this works, it does leave the code very disjointed and difficult to follow. Typically you only have a single event handler in a given class.

**Pull request closed without merging**